chore(*): update minimum Yarn version, do some Yarn-related cleanups

mgol opened the pull request:
<!-- General PR submission guidelines https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#submit-pr -->
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Build change.

**What is the current behavior? (You can also link to an open issue here)**

See the next section.

**What is the new behavior (if this is a feature change)?**

Included changes:

*chore(\*): update minimum Yarn version from 1.3.2 to 1.10.1*

Yarn 1.10 added the integrity field to the lockfile, making newer Yarn users
have their lockfile changed a lot if they run `yarn`. This commit updates the
required Yarn version to be at least 1.10.1 and changes Travis & Jenkins to use
Yarn 1.10.1

*chore(\*): change the package.json's engines grunt field to grunt-cli*

The grunt field suggested it's the grunt package version we're checking while
we check the grunt-cli version instead.

*chore(\*): stop separating Yarn script arguments from script names via " -- "*

The " -- " separator is necessary in npm but not in Yarn. In fact, it's
deprecated in Yarn and some future version is supposed to start passing this
parameter directly to the scripts which may break them.

*chore(\*): Don't install grunt-cli globally during the build*

It's enough to use `yarn grunt` instead of `grunt` and the global grunt-cli
installation is no longer needed.

**Does this PR introduce a breaking change?**

No.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our [guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- ~~Fix/Feature: [Docs](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#documentation) have been added/updated~~
- ~~Fix/Feature: Tests have been added; existing tests pass~~

**Other information**:

mgol committed:
chore(*): update minimum Yarn version from 1.3.2 to 1.10.1

Yarn 1.10 added the integrity field to the lockfile, making newer Yarn users
have their lockfile changed a lot if they run `yarn`. This commit updates the
required Yarn version to be at least 1.10.1 and changes Travis & Jenkins to use
Yarn 1.10.1

mgol committed:
chore(*): change the package.json's engines grunt field to grunt-cli

The grunt field suggested it's the grunt package version we're checking while
we check the grunt-cli version instead.

mgol committed:
chore(*): stop separating Yarn script arguments from script names via " -- "

The " -- " separator is necessary in npm but not in Yarn. In fact, it's
deprecated in Yarn and some future version is supposed to start passing this
parameter directly to the scripts which may break them.

mgol committed:
chore(*): don't install grunt-cli globally during the build

It's enough to use `yarn grunt` instead of `grunt` and the global grunt-cli
installation is no longer needed.

gkalpak reviewed (approved):
:tada:

petebacondarwin commented on package.json:
```diff
@@ -10,8 +10,8 @@
   },
   "engines": {
     "node": "^8.9.1",
```
Can we change the node engine to be more lenient to newer versions of node too?

mgol replied:
Sure! Do we want to upgrade to Node 10 or stay at 8, just newer?

petebacondarwin replied:
I think we should make it relaxed enough to accept any engines that work with our build. So perhaps `>= 8.9.1 < 11`?

mgol replied:
@petebacondarwin We don't have a way to verify if it keeps working at a version we don't work on. Also, `.nvmrc` needs to specify one version, we can't use 2 at the same time. So if we want to allow newer versions maybe it'd be better to say `>=8.12.0` so that all newer versions are accepted?

petebacondarwin replied:
Happy to go with `>=8.12.0`

petebacondarwin commented on scripts/jenkins/build.sh:
```diff
@@ -21,21 +21,21 @@ rm -f angular.js.size
 
 
 # BUILD #
-yarn run grunt -- ci-checks package --no-color
```
❤️

mgol replied:
That reminds me I can shorten it to `yarn grunt` instead of `yarn run grunt`... We already use the former form in a few places.

mgol committed:
chore(*): use `yarn grunt` instead of `yarn run grunt`

The former form is shorter.

mgol committed:
chore(*): don't define the `grunt` Yarn script

As opposed to npm, `yarn binName` invokes a binary named `binName` exposed
by the respective package so the `grant` Yarn script is no longer needed.

mgol committed:
chore(*): allow Node versions newer than 8; bump the minimum

Narretz commented:
Just a note: when running yarn, I still get `warning angularjs@: The engine "grunt-cli" appears to be invalid.` I think we can just throw it out.

mgol commented:
@Narretz we parse the `engines` fields ourselves for validation but Yarn does it as well. Those two are unrelated.
